### PR TITLE
Feature/ab#957 adds licenses vuln for python

### PR DESCRIPTION
**Pull request by Anthony-Mckale**
What:
python license data only comes when the cyclonedx source is the local environment

How:
- download pypi data for packages
- extract license

Stretch:
- extract vulnerability data from pypi
- download cve information
- add into cyclonedx

What:

- added pypi data to eze/data/xxx.json
   (lookup table for pypi licenses to spdx licenses)
- renamed eze.utils.sbom to more descriptive eze.utils.license
- extended eze/plugins/tools/python_cyclonedx.py to do SCA, and populate license using new eze.utils.pypi::get_pypi_package_data method
- re-written eze.utils.cve util to use NVD data (more official source)
   https://services.nvd.nist.gov/rest/json/cve/1.0/CVE-2013-5123
- created eze/utils/purl.py to parse purl formats https://github.com/package-url/purl-spec
- created eze/utils/pypi.py to pull license / vulnerability data
   https://pypi.org/pypi/aws-encryption-sdk/1.2.0/json

MINOR:
- adding linting for unused imports etc
- fixed poor spelling of "advisory" was "advisitory"
- created cyclonedx imrpovement ticket 

NOTES:
- new cyclonedx plugin effectively can do everything safety can, and with extension everything pip-rot could (has all the version data in the pypi feed)

(originally started as [AB#847](https://dev.azure.com/riversafe/0b0abc1f-f46a-4d96-a470-e1c546e3cf29/_workitems/edit/847))

[AB#957](https://dev.azure.com/riversafe/0b0abc1f-f46a-4d96-a470-e1c546e3cf29/_workitems/edit/957)

### CODE REVIEW

File: eze/plugins/tools/node_npmaudit.py [hunk 1]
```diff
@@ -184,10 +184,10 @@ def parse_npm_v7_report(self, parsed_json: dict) -> ScanResult:
                 "file_location": None,
             }
 
-            # WARNING: AB-524: limitation, for know just showing first advisitory
-            advisitory_source = py_.get(vulnerability, "via[0].source", False)
-            if advisitory_source:
-                vulnerability_vo["identifiers"]["npm"] = advisitory_source
+            # WARNING: AB-524: limitation, for know just showing first advisory
```
Comment (mgonzales-rs) [inline eze/plugins/tools/node_npmaudit.py:187]: lil observation: for **known** or for **now** ?

**Review by mgonzales-rs — COMMENTED** [review]
It looks good, just got a warning message for python safety 
https://services.nvd.nist.gov/rest/json/cve/1.0/CVE-2013-5123

image

File: eze/plugins/tools/python_cyclonedx.py [hunk 2]
```diff
@@ -88,4 +97,26 @@ async def run_scan(self) -> ScanResult:
...
+            if not purl_breakdown or purl_breakdown.type != "pypi":
+                continue
+            pypi_data: PypiPackageVO = get_pypi_package_data(purl_breakdown.name, purl_breakdown.version)
+            vulnerabilities.extend(pypi_data.vulnerabilities)
+            warnings.extend(pypi_data.warnings)
+            licenses = component.get("licenses", [])
+            if len(licenses) == 0:
+                for pypi_license in pypi_data.licenses:
+                    licenses.append({"license": {"name": pypi_license}})
+                component["licenses"] = licenses
```
[the remark this answers is not in the document]
Comment (Anthony-Mckale) [reply eze/plugins/tools/python_cyclonedx.py:118]: absolutely nothing, all our code only reads the first one

technically they shouldn't have multiple licenses :)

File: eze/plugins/tools/node_npmaudit.py (hunk 1 above)
Comment (Anthony-Mckale) [reply eze/plugins/tools/node_npmaudit.py]: now

File: eze/plugins/tools/python_cyclonedx.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Anthony-Mckale) [reply eze/plugins/tools/python_cyclonedx.py:118]: ps the safety issue shouldn't happen, will commit and see what goats do

**Pull request merged**